service/src/main/java/com/codeforcommunity/requester/S3Requester.java

```diff
@@ -273,4 +275,33 @@ public static String uploadHTML(
 
     return String.format("%s/%s/%s", externs.getBucketPublicUrl(), directoryName, name);
   }
+
+  /**
+   * Validate the given string encoding of HTML and overwrite an existing file in the user upload S3 bucket.
+   *
```

Comment:
💭 If the add template route already exists and overwrites templates with the same name, do we still need this route? Not totally sure how the UI will work for creating/updating templates, but I feel like we could use the add template route for both functions, considering the only difference here is that we first check if the file exists.

Reply:
Good idea-- I was also thinking about this and agree that they might be redundant. Since load, delete, and edit all check for the existence of the file, I could abstract this out into a separate route to be called before each one (therefore eliminating the edit template) but not sure if this is the best design.

Reply:
Hmm I agree this route is redundant and we can probably close this PR. Sorry about that!

If the load and delete routes both check for the file's existence, we can probably just make a private helper method that does this and have both routes call that method. It's great that you noticed this tho!